quickwit/quickwit-search/src/jaeger_collector.rs

```diff
@@ -0,0 +1,215 @@
+    }
+
+    pub fn term_dict_field_names(&self) -> HashSet<String> {
+        HashSet::from_iter([self.trace_id_field_name.clone()])
+    }
+}
+
+impl Collector for FindTraceIdsCollector {
+    type Fruit = Vec<TraceIdSpanTimestamp>;
+    type Child = FindTraceIdsSegmentCollector;
+
+    fn for_segment(
+        &self,
+        _segment_local_id: u32,
+        segment_reader: &SegmentReader,
+    ) -> tantivy::Result<Self::Child> {
+        let trace_id_ff_reader = segment_reader
+            .fast_fields()
+            .u64s(&self.trace_id_field_name)?;
+        let span_timestamp_column = segment_reader
+            .fast_fields()
+            .date(&self.span_timestamp_field_name)?;
+
+        let trace_id_field = segment_reader
+            .schema()
+            .get_field(&self.trace_id_field_name)?;
+        let inverted_index_reader = segment_reader.inverted_index(trace_id_field)?;
+
+        Ok(Self::Child {
+            num_traces: self.num_traces,
+            trace_id_ff_reader,
+            span_timestamp_column,
+            inverted_index_reader,
+            buckets: FnvHashMap::default(),
+        })
+    }
+
+    fn merge_fruits(
+        &self,
+        segment_fruits: Vec<<Self::Child as SegmentCollector>::Fruit>,
+    ) -> tantivy::Result<Self::Fruit> {
+        let mut buckets = FnvHashMap::default();
+        for segment_fruit in segment_fruits {
+            for fruit in segment_fruit {
+                buckets
+                    .entry(fruit.trace_id)
+                    .and_modify(|entry| {
+                        if *entry < fruit.span_timestamp {
+                            *entry = fruit.span_timestamp
+                        }
+                    })
+                    .or_insert(fruit.span_timestamp);
+            }
+        }
+        let merged_fruits = buckets
+            .into_iter()
+            .sorted_by_key(|(_, span_timestamp)| *span_timestamp)
+            .rev()
+            .take(self.num_traces)
+            .map(|(trace_id, span_timestamp)| TraceIdSpanTimestamp::new(trace_id, span_timestamp))
+            .collect();
+        Ok(merged_fruits)
+    }
+
+    fn requires_scoring(&self) -> bool {
+        false
+    }
+}
+
+/// This a naive implementation of a top k algorithm with duplicates. A more robust implementation
+/// will follow.
+pub struct FindTraceIdsSegmentCollector {
+    num_traces: usize,
+    trace_id_ff_reader: MultiValuedFastFieldReader<u64>,
+    span_timestamp_column: Arc<dyn Column<DateTime>>,
+    inverted_index_reader: Arc<InvertedIndexReader>,
+    buckets: FnvHashMap<TraceIdTermOrd, SpanTimestamp>,
+}
+
+impl FindTraceIdsSegmentCollector {
+    fn trace_id_term_ord(&self, doc: DocId) -> TraceIdTermOrd {
+        self.trace_id_ff_reader
+            .get_first_val(doc)
+            .expect("There should be exactly one trace ID per span.")
+    }
+
+    fn trace_id(&self, trace_id_term_ord: TraceIdTermOrd) -> TraceId {
+        let mut trace_id = vec![0u8; 16];
```

Comment:
You probably wanted to do 

```rust
        let mut trace_id = Vec::with_capacity(16);
```

but I wonder if this is the right idea here?
If you are certain that the trace id is 16 bytes long, then the best is probably to work off the stack 
and remove allocations.

Reply:
> You probably wanted to do

Courtesy of Copilot :D

> but I wonder if this is the right idea here?

Good point. Will fix.